File: Mar 8 141. Linked List Cycle (Easy)/Solution.py
class Solution:
    def hasCycle(self, head: Optional[ListNode]) -> bool:
        if head is None or head.next is None:
            return False
        slow = head
        fast = head.next
        while fast.next and fast.next.next and slow != fast:
            slow = slow.next
            fast = fast.next.next
        return slow == fast

File: Mar 8 141. Linked List Cycle (Easy)/Solution.go
func hasCycle(head *ListNode) bool {
    if head == nil || head.Next == nil{
        return false
    }
    slow := head
    fast := head.Next
    
    for fast.Next != nil && fast.Next.Next != nil && fast != slow{
        slow = slow.Next
        fast = fast.Next.Next
    }
    
    return slow == fast
}
